lua/autorun/client/TrueCulling.lua

```diff
@@ -0,0 +1,68 @@
+jit.on()
+
+CULLFRONT = math.sin(0.57357644) -- Degrees forward angle in sine
+
+function main()
+	allEnts = ents.GetAll()
+	if engine.TickCount()%4 == 0 then
+		for a, ents_ in pairs(allEnts) do
+			if engine.TickCount()%10 == 0 then
+				ents_:SetNoDraw(true)
+			end
+			if engine.TickCount()%10 == 0 then
+				if ents_:GetNoDraw(true) then
+					if util.IsPointInCone(ents_:GetPos(), LocalPlayer():GetPos(), LocalPlayer():GetAimVector(), CULLFRONT, 100000) then
+						if LocalPlayer():IsLineOfSightClear(ents_:GetPos() + Vector(-25,-25,-25)) or LocalPlayer():IsLineOfSightClear(ents_:GetPos() + Vector(10,10,10)) then
```

Comment:
Can you explain what this is supposed to do? It seems like you are checking tracing visibility to two arbitrary points near the object with no regard to the size of that object.

Reply:
I used to have a script that did what I think you are trying to do here, where it would choose a random point within the object bounding box, and it would only hide the object if it failed all of the past 10 or so checks. That way it would basically stochastically estimate the visibility of the entire object. I later removed it because it was stupid.